File: 11-container-with-most-water/container-with-most-water.php
class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $n = sizeof($height);
        $ans = 0;
        $l = 0; $r = $n-1;
        while($l < $r){
            $mn = min($height[$l], $height[$r]);
            $ans = max($ans, $mn*($r-$l));
            $height[$l] > $height[$r] ? $r-- : $l++;
        }
        return $ans;
    }
}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();

        int l = 0, r = n-1;
        int ans = 0;
        while(l < r){
            ans = max(ans, min(height[l],height[r])*(r-l));
            height[r] < height[l] ? r-- : l++;
        }

        return ans;
    }
};